README-fr.md

```diff
@@ -154,195 +154,199 @@ Lisez les [directives de contribution](CONTRIBUTING.md).
 
-### Step 1: Outline use cases, constraints, and assumptions
+### Etape 1 : Dresser les cas d'usage, les contraintes et les hypothèses
 
-Gather requirements and scope the problem.  Ask questions to clarify use cases and constraints.  Discuss assumptions.
+Recueillez les besoins et cadrez le problème. Posez des questions pour clarifier les cas d'usage et les contraintes. Discutez des hypothèses.
 
-* Who is going to use it?
-* How are they going to use it?
-* How many users are there?
-* What does the system do?
-* What are the inputs and outputs of the system?
-* How much data do we expect to handle?
-* How many requests per second do we expect?
-* What is the expected read to write ratio?
+* Qui vont l'utiliser?
+* Comment vont-ils l'utiliser?
+* Combien y aura-t-il d'utilisateurs?
+* Que fait le système?
+* Quelles sont les entrées-sorties du système?
+* Quelle est la quantité de données qu'on s'attend à traiter?
+* Combien de requêtes par seconde qu'on s'attend à avoir?
+* Quel est le ratio lecture/écriture qu'on s'attend à avoir?
 
-### Step 2: Create a high level design
+### Etape 2: Créer une conception de haut niveau
 
-Outline a high level design with all important components.
+Faites une ébauche de conception de haut niveau avec tous les composants importants.
 
-* Sketch the main components and connections
-* Justify your ideas
+* Esquissez les composants principaux et leurs connexions
+* Justifiez vos idées
 
-### Step 3: Design core components
+### Etape 3: Concevoir les composants essentiels
 
-Dive into details for each core component.  For example, if you were asked to [design a url shortening service](solutions/system_design/pastebin/README.md), discuss:
+Entrez dans les détails de chaque composant essentiel. Par exemple, si on vous demande de [concevoir un service de raccourcissement d'URL](solutions/system_design/pastebin/README.md), discutez de : 
 
-* Generating and storing a hash of the full url
-    * [MD5](solutions/system_design/pastebin/README.md) and [Base62](solutions/system_design/pastebin/README.md)
-    * Hash collisions
-    * SQL or NoSQL
-    * Database schema
-* Translating a hashed url to the full url
-    * Database lookup
-* API and object-oriented design
+* Générer et conserver un hachage de l'URL complète
+    * [MD5](solutions/system_design/pastebin/README.md) et [Base62](solutions/system_design/pastebin/README.md)
+    * Collisions de hachage
+    * SQL ou NoSQL
+    * Schéma de base de données
+* Traduire l'URL de hachage en URL complète
+    * Recherche en base de données
+* API et conception orientée objet
 
-### Step 4: Scale the design
+### Etape 4: Mettre à l'echelle la conception
 
-Identify and address bottlenecks, given the constraints.  For example, do you need the following to address scalability issues?
+Etant donné les contraintes, identifiez et éliminez les goulets d'étranglement. Par exemple pour régler des problèmes de *scalabilité*, avez-vous besoin de?

```

Comment:
```suggestion
Etant donné les contraintes, identifiez et éliminez les goulots d'étranglement. Par exemple pour régler des problèmes de *scalabilité*, avez-vous besoin de ?

```

Reply:
Both are correct. I think we can say `goulet` or `goulot`

Reply:
I just checked and you're right, let's keep it like this :wink: